debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/ModularityHelper.java

```diff
@@ -0,0 +1,172 @@
+	 * outtputter to print details about modules
+	 * matched by filter.
+	 * 
+	 * @param		out			The printstream that will be provided to outputter.
+	 * @param		filter		Used to determine whether a module will be output.
+	 * 							Will be run on all loaded modules.
+	 * @param		outputter	Used to output details about a module that was
+	 * 							matched by the filter. Will be run on any module that
+	 * 							filter returns true for.
+	 * @param		filterArg	A string that is to be used as additional data for filter.
+	 * 							Will be passed to filter everytime filter is called.
+	 */
+	public static void iterateModules(PrintStream out, ModuleIteratorFilter filter, ModuleOutput outputter, String filterArg) throws CorruptDataException {
+		J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+		if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+			GCClassLoaderIterator iterator = GCClassLoaderIterator.from();
+			while (iterator.hasNext()) {
+				J9ClassLoaderPointer classLoaderPointer = iterator.next();
+				HashTable<J9ModulePointer> moduleHashTable = ModuleHashTable
+						.fromJ9HashTable(classLoaderPointer.moduleHashTable());
+				SlotIterator<J9ModulePointer> slotIterator = moduleHashTable.iterator();
+				while (slotIterator.hasNext()) {
+					J9ModulePointer modulePtr = slotIterator.next();
+					if (filter.filter(modulePtr, filterArg)) {
+						outputter.print(modulePtr, out);
+					}
+				}
+			}
+		}
+	}
+
+
+	/**
+	 * Traverses through all loaded modules.
+	 * Uses outtputter to print details about modules
+	 * matched by filter.
+	 * 
+	 * @param		out			The printstream that will be provided to outputter
+	 * @param		filter		Used to determine whether a module will be output.
+	 * 							Will be run on all loaded modules.
+	 * @param		outputter	Used to output details about a module that was matched
+	 * 							by the filter. Will be run on any module that filter
+	 * 							returns true for.
+	 * @param		filterArg	A string that is to be used as additional data for filter.
+	 * 							Will be passed to filter everytime filter is called.
+	 */
+	public static void iteratePackages(PrintStream out, PackageIteratorFilter filter, PackageOutput outputter, String filterArg) throws CorruptDataException {
+		J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
+		if (JavaVersionHelper.ensureJava9AndUp(vm, out)) {
+			GCClassLoaderIterator iterator = GCClassLoaderIterator.from();
+			while (iterator.hasNext()) {
+				J9ClassLoaderPointer classLoaderPointer = iterator.next();
+				HashTable<J9PackagePointer> packageHashTable = PackageHashTable
+						.fromJ9HashTable(classLoaderPointer.packageHashTable());
+				SlotIterator<J9PackagePointer> slotIterator = packageHashTable.iterator();
+				while (slotIterator.hasNext()) {
+					J9PackagePointer packagePtr = slotIterator.next();
+					if (filter.filter(packagePtr, filterArg)) {
+						outputter.print(packagePtr, out);
+					}
+				}
+			}
+		}
+	}
+}
```

Comment:
Please don't use internal tabs (those other than the group at the beginning of a line).
All source files should end with a final newline.

Reply:
All internal tabs within `ModularityHelper.java` and `FindModulesCommand.java` have been replaced by spaces. Missing end newline has been added.